src/Provider/ProviderInterface.php

```diff
@@ -61,7 +61,7 @@ public function authorize(array $options = []);
      * @param array $params
      * @return AccessToken
      */
-    public function getAccessToken($grant = 'authorization_code', array $params = []);
+    public function getAccessToken($grant, array $params = []);
```

Comment:
We are currently accepting either string or `GrantInterface`, having the `GrantFactory` take care of that for us. Doesn't seem nice to *require* and instantiated `GrantInterface`.